### PR TITLE
v04 of the draft

### PR DESCRIPTION
**Pull request by mjethanandani**

### CODE REVIEW

File: draft/draft-ietf-babel-yang-model.xml [hunk 1]
```diff
@@ -173,22 +173,44 @@ INSERT_TEXT_FROM_FILE(../src/yang/ietf-babel@YYYY-MM-DD-abridged-tree.txt)
...
+        subsystems. By augmenting the module it provides a common building
+        block for routes, and Routing Information Bases (RIBs). It also has a
+        reference to an interface defined by <xref target="RFC8343">A YANG
+        Data Model for Interface Management</xref>.</t>
+
+        <t>A router running Babel routing protocol can determine the
+        parameters it needs to use for an interface based on the interface
+        name. For example, it can detect that eth0 is a wired interface, and
+        that wlan0 is a wireless interface. This is not true for a tunnel
+        interface, where the link parameters need to configured explicitly.
```
Comment (bhstark2) [inline draft/draft-ietf-babel-yang-model.xml]: s/need to configured/need to be configured/

File: draft/draft-ietf-babel-yang-model.xml [hunk 2]
```diff
@@ -173,22 +173,44 @@ INSERT_TEXT_FROM_FILE(../src/yang/ietf-babel@YYYY-MM-DD-abridged-tree.txt)
...
+        </t>
+
+        <t>For a wired interface, it will assume '2-out-of-3'
+        'metric-algorithm', and 'split-horizon' set to true. On other other
```
Comment (bhstark2) [inline draft/draft-ietf-babel-yang-model.xml]: s/on other other/on the other/

File: src/yang/ietf-babel.yang [hunk 3]
```diff
@@ -437,50 +453,50 @@ module ietf-babel {
...
+          type identityref {
+            base metric-comp-algorithms;
+          }
+          mandatory true;
+          description
             "Indicates the metric computation algorithm used on this
-             interface. The value MUST be one of those listed in
-             'metric-comp-algorithms'.";
+             interface. The value MUST be one of those identities
+             based 'metric-comp-algorithms'.";
```
Comment (bhstark2) [inline src/yang/ietf-babel.yang]: s/based/based on/ // I double-checked the email thread to make sure it said "based on"

File: src/yang/ietf-babel.yang [hunk 4]
```diff
@@ -545,48 +560,47 @@ module ietf-babel {
...
+          description
+            "List of supported certificate types, in order of
              preference. The values MUST be among those listed in the
-             babel-dtls-cert-types parameter. This list is used to
-             populate the server_certificate_type extension in a
-             Client Hello.  Values that are present in at least one
-             instance in the babel-dtls-certs object of a referenced
-             babel-dtls instance and that have a non-empty
-             babel-cert-private-key will be used to populate the
+             types parameter in the list of certs under dtls. This
```
Comment (bhstark2) [inline src/yang/ietf-babel.yang]: No, babel-dtls-cert-types (old) should simply become dtls-cert-types, which is a parameter in the root object:
       leaf-list dtls-cert-types {
         type identityref {
           base dtls-cert-types;

s/types parameter in the list of certs under dtls/dtls-cert-types parameter/

File: src/yang/ietf-babel.yang [hunk 5]
```diff
@@ -596,32 +610,38 @@ module ietf-babel {
...
-	  type inet:uri;
-	  config false;
+          type inet:uri;
+          config false;
           description
-	    "A reference or url link to a file that contains a
+            "A reference or url link to a file that contains a
              timestamped log of packets received and sent on
-             babel-udp-port on this interface. The [libpcap] file
+             udp port on this interface. The [libpcap] file
```
Comment (bhstark2) [inline src/yang/ietf-babel.yang]: It looks like the parameter name in this model is udp-port. I think the hyphen is needed.
s/udp port/udp-port/

File: src/yang/ietf-babel.yang [hunk 6]
```diff
@@ -669,11 +689,17 @@ module ietf-babel {
...
 	  action reset {
 	    description
-              "The information model [RFC ZZZZ] defines this reset
-               action as a system-wide reset of Babel statistics
-               parameters, but in YANG the reset action has to be
-               contained in the container where the action needs to
-               be performed.";
+              "The information model [RFC ZZZZ] defines reset
+               action as a system-wide reset of Babel statistics.
+               In YANG the reset action is associated with
```
Comment (bhstark2) [inline src/yang/ietf-babel.yang]: s/with/with the/

**Review by bhstark2 — CHANGES_REQUESTED** [review]

File: draft/draft-ietf-babel-yang-model.xml (hunk 1 above)
Comment (mjethanandani) [reply draft/draft-ietf-babel-yang-model.xml]: Ok

File: draft/draft-ietf-babel-yang-model.xml (hunk 2 above)
Comment (mjethanandani) [reply draft/draft-ietf-babel-yang-model.xml]: Ok

File: src/yang/ietf-babel.yang (hunk 3 above)
Comment (mjethanandani) [reply src/yang/ietf-babel.yang]: Ok.

File: src/yang/ietf-babel.yang (hunk 4 above)
Comment (mjethanandani) [reply src/yang/ietf-babel.yang]: s/types parameter in the list of certs under dtls/dtls-cert-types/

File: src/yang/ietf-babel.yang (hunk 5 above)
Comment (mjethanandani) [reply src/yang/ietf-babel.yang]: Ok

**Review by bhstark2 — APPROVED** [review]

**Pull request merged**